File: Server_Go/routes/pathRoutes.go
package routes

import (
	"github.com/gorilla/mux"
)

func RegisterPathRoutes(r *mux.Router) {
	r.HandleFunc("/", pathController.getPath).Name("Path Info")
	r.HandleFunc("/company", pathController.getPathsByCompany).Name("Get Paths By Company")
	r.HandleFunc("/create", pathController.createPath).Name("Create Path")
	r.HandleFunc("/update", pathController.updatePath).Name("Update Path")
}

File: Server/routes/PathRoutes.js
const express = require('express');
const router = express.Router();
const pathController = require('../controllers/pathController.js');
const isAuthenticated = require('../Utils/AuthenticationUtil.js');
 
router.get('/', isAuthenticated, pathController.getPath);

router.get('/company', isAuthenticated, pathController.getPathsByCompany);

router.post('/create', isAuthenticated, pathController.createPath);

router.post('/update', isAuthenticated, pathController.updatePath);

module.exports = router;
